src/pages/Wallet/NewAddressPage.tsx

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright 2018 - 2022 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import addressToGroup from 'alephium-js/dist/lib/address'
+import { TOTAL_NUMBER_OF_GROUPS } from 'alephium-js/dist/lib/constants'
+import { deriveNewAddressData } from 'alephium-js/dist/lib/wallet'
+import { FC, useCallback, useEffect, useState } from 'react'
+import styled from 'styled-components'
+
+import ExpandableSection from '../../components/ExpandableSection'
+import ColoredLabel from '../../components/Inputs/ColoredLabel'
+import { getRandomColor } from '../../components/Inputs/ColorPicker'
+import KeyValueInput from '../../components/Inputs/InlineLabelValueInput'
+import Select from '../../components/Inputs/Select'
+import Toggle from '../../components/Inputs/Toggle'
+import { ModalFooterButton, ModalFooterButtons } from '../../components/Modal'
+import HorizontalDivider from '../../components/PageComponents/HorizontalDivider'
+import { Section } from '../../components/PageComponents/PageContainers'
+import { useGlobalContext } from '../../contexts/global'
+import { useModalContext } from '../../contexts/modal'
+
+let NewAddressPage: FC<{ className?: string }> = ({ className }) => {
+  const [addressLabel, setAddressLabel] = useState({ title: '', color: getRandomColor() })
+  const [isMainAddress, setIsMainAddress] = useState(false)
+  const [newAddressData, setNewAddressData] = useState<{ address: string; addressIndex: number }>()
+  const [newAddressGroup, setNewAddressGroup] = useState<number>()
+  const { wallet, addressesInfo, saveAddressInfo } = useGlobalContext()
+  const currentAddressIndexes = addressesInfo.map(({ index }) => index)
+  const currentMainAddressInfo = addressesInfo.find(({ isMain }) => isMain)
+  const { onModalClose } = useModalContext()
+
+  const generateNewAddress = useCallback(
+    (group?: number) => {
+      if (wallet?.seed) {
+        const data = deriveNewAddressData(wallet.seed, group, undefined, currentAddressIndexes)
+        setNewAddressData(data)
+        setNewAddressGroup(group || addressToGroup(data.address, TOTAL_NUMBER_OF_GROUPS))
+      }
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
I specifically **DONT** want `currentAddressIndexes` to be listed here...

Reply:
In that case, maybe use a `ref` (`useRef`) to store the `initialAddressIndexes`, that you would then use in `deriveNewAddressData` ? This would avoid overriding the eslint rule while also avoiding refreshing the address index list.